src/v/iceberg/table_requests.h

```diff
@@ -0,0 +1,40 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "container/chunked_hash_map.h"
+#include "container/fragmented_vector.h"
+#include "iceberg/partition.h"
+#include "iceberg/schema.h"
+#include "iceberg/table_requirement.h"
+#include "iceberg/table_update.h"
+
+namespace iceberg {
+
+struct create_table_request {
+    ss::sstring name;
+    schema schema;
+    std::optional<ss::sstring> location;
```

Comment:
Mechanically, because it's not listed as required here https://github.com/apache/iceberg/blob/5439cbdb278232779fdd9a392bbf57f007f9bda0/open-api/rest-catalog-open-api.yaml#L3326-L3347

Conceptually I believe this is optional so that a create table statement can be spooled across multiple requests, using the stage-create option.